Content.Server/Shuttles/Systems/StationAnchorSystem.cs

```diff
@@ -64,6 +75,60 @@ private void OnAnchorStationChange(Entity<StationAnchorComponent> ent, ref Ancho
             SetStatus(ent, false);
     }
 
+    // Frontier: All these functions are used to handle device linking for anchors.
+    private void OnInit(EntityUid uid, StationAnchorComponent anchor, ComponentInit args)
+    {
+        _signalSystem.EnsureSinkPorts(uid, anchor.OnPort, anchor.OffPort, anchor.TogglePort);
+    }
+
+    private void OnPacketReceived(EntityUid uid, StationAnchorComponent component, DeviceNetworkPacketEvent args)
+    {
+        if (!args.Data.TryGetValue(DeviceNetworkConstants.Command, out string? command) ||
+            command != DeviceNetworkConstants.CmdSetState)
+            return;
+        if (!args.Data.TryGetValue(DeviceNetworkConstants.StateEnabled, out bool enabled))
+            return;
+
+        SetAnchorPower((uid, component), enabled);
+    }
+
+    private void OnSignalReceived(EntityUid uid, StationAnchorComponent component, ref SignalReceivedEvent args)
+    {
+        if (args.Port == component.OffPort)
+        {
+            SetAnchorPower((uid, component), false);
+        }
+        else if (args.Port == component.OnPort)
+        {
+            SetAnchorPower((uid, component), true);
+        }
+        else if (args.Port == component.TogglePort)
+            ToggleAnchorPower((uid, component));
+    }
+
+    private void SetAnchorPower(Entity<StationAnchorComponent> ent, bool value)
+    {
+        if (!EntityManager.TryGetComponent<PowerChargeComponent>(ent, out var entPowerHandler))
+        {
+            Console.WriteLine($"Unable to find PowerChargeComponent in StationAnchor. ID:{ent.Owner}");
+            return;
+        }
+
+        _chargeSystem.SetSwitchedOn(ent, entPowerHandler, value);
+    }
+
+    private void ToggleAnchorPower(Entity<StationAnchorComponent> ent)
+    {
+        if (!EntityManager.TryGetComponent<PowerChargeComponent>(ent, out var entPowerHandler))
+        {
+            Console.WriteLine($"Unable to find PowerChargeComponent in StationAnchor. ID:{ent.Owner}");
+            return;
+        }
+
+        _chargeSystem.SetSwitchedOn(ent, entPowerHandler, (!entPowerHandler.SwitchedOn));
```

Comment:
```suggestion
        _chargeSystem.SetSwitchedOn(ent, entPowerHandler, !entPowerHandler.SwitchedOn);
```
Unneeded parentheses.